### ISSUE TITLE
Bug with Ecobee update 

### ISSUE THREAD

**Issue by marcolivierarsenault** [marcolivierarsenault/tefnut#52]
Realised that when the server sends us data, it does not mean the data is fresh.

Got a bug where humidifier ran for 1 hour becasue I kept getting the same old data.

Freshness is already available in the ecobee object we are getting back:

{"severity": "DEBUG", "module": "ecobee", "msg": "EcobeeThermostatResponse(page=Page(page=1, totalPages=1, pageSize=1, total=1), thermostatList=[Thermostat(identifier='522674392583', name='My ecobee', thermostat_rev='230205220646', is_registered=True, model_number='vulcanSmart', brand='ecobee', features='Home,HomeKit', last_modified='2023-02-05 22:06:46', thermostat_time='2023-02-05 20:01:11', utc_time='2023-02-06 01:01:11', audio=None, alerts=None, reminders=None, settings=None, runtime=Runtime(runtime_rev='230206010027', connected=True, first_connected='2021-10-05 02:18:34', connect_date_time='2023-02-05 23:56:56', disconnect_date_time='2023-02-05 16:30:04', last_modified='2023-02-06 01:00:27', **last_status_modified='2023-02-06 01:00:27'**, runtime_date='2023-02-06', runtime_interval=9, actual_temperature=683, actual_humidity=36, raw_temperature=683, show_icon_mode=0, desired_heat=680, desired_cool=716, desired_humidity=36, desired_dehumidity=54, desired_fan_mode='on', actual_voc=-5002, actual_co2=-5002, actual_aq_accuracy=0, actual_aq_score=-5002, desired_heat_range=[450, 790], desired_cool_range=[650, 920]), extended_runtime=None, electricity=None, devices=None, location=None, energy=None, technician=None, utility=None, management=None, weather=None, events=None, program=None, house_details=None, oem_cfg=None, equipment_status=None, notification_settings=None, privacy=None, version=None, security_settings=None, filter_subscription=None, remote_sensors=None)], status=Status(code=0, message=))"}

After looking at the logs, you can see the thermostat time go up, but last status staying still:

Query from Loki:
```
{application="tefnut-prod"} | json | module = `ecobee` | regexp ".*thermostat_time='(?P<thermotime>\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}).*last_status_modified='(?P<updatetime>\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}).*" | line_format "{{.thermotime}} {{.updatetime}}"
```

Log time | thermostat_time |  last_status_modified
-- | -- | --
 2023-02-05 19:51:11 | 2023-02-05 19:51:11| 2023-02-06 00:50:27 |  
 2023-02-05 19:46:11 | 2023-02-05 19:46:11| 2023-02-06 00:43:55 |  
 2023-02-05 19:41:11 | 2023-02-05 19:41:11| 2023-02-06 00:40:55 |  
 2023-02-05 19:40:11 | 2023-02-05 19:40:11| 2023-02-06 00:39:12 |  
 2023-02-05 19:39:11 | 2023-02-05 19:39:11| **2023-02-06 00:37:10** |  
 2023-02-05 19:38:11 | 2023-02-05 19:38:11| **2023-02-06 00:37:10** |  
 2023-02-05 19:37:11 | 2023-02-05 19:37:11| **2023-02-06 00:37:10** |  
 2023-02-05 19:36:11 | 2023-02-05 19:36:11| **2023-02-06 00:33:40** |  
 2023-02-05 19:35:11 | 2023-02-05 19:35:11| **2023-02-06 00:33:40** |  
 2023-02-05 19:34:11 | 2023-02-05 19:34:11| **2023-02-06 00:33:40** |  
 2023-02-05 19:33:11 | 2023-02-05 19:33:11| **2023-02-06 00:30:28** |  
 2023-02-05 19:32:11 | 2023-02-05 19:32:11| **2023-02-06 00:30:28** |  
 2023-02-05 19:31:11 | 2023-02-05 19:31:11| **2023-02-06 00:30:28** |  
 2023-02-05 19:30:11 | 2023-02-05 19:30:11| **2023-02-06 00:27:25** |  
 2023-02-05 19:29:11 | 2023-02-05 19:29:11| **2023-02-06 00:27:25** |  
 2023-02-05 19:28:11 | 2023-02-05 19:28:11| **2023-02-06 00:27:25** |  
 2023-02-05 19:23:11 | 2023-02-05 19:23:11| 2023-02-06 00:21:25 |  
 2023-02-05 19:18:11 | 2023-02-05 19:18:11| 2023-02-06 00:17:25 |  
 2023-02-05 19:13:12 | 2023-02-05 19:13:11| 2023-02-06 00:12:40|

---

Strategy:
Dont assume freshness, validate freshness,  only use data if fresh

**Comment by marcolivierarsenault** [comment]
Example of the issue

![image](https://user-images.githubusercontent.com/2634090/222869482-f5c5da37-fb2f-42c9-8370-d1fe9e00d575.png)

![image](https://user-images.githubusercontent.com/2634090/222869493-27698c70-623d-49b0-b718-c1cc7654d422.png)

**Comment by marcolivierarsenault** [comment]
fixed in #61
